assets/js/modules/analytics/common/account-select.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Analytics Account Select component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { Select, Option } from '../../../material-components';
+import { STORE_NAME, ACCOUNT_CREATE } from '../datastore/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function AccountSelect() {
+	const accounts = useSelect( ( select ) => select( STORE_NAME ).getAccounts() ) || [];
```

Comment:
It's only there to prevent an error in the case that it's `undefined`. This value is only used to populate options below which requires an array. I could move this fallback to that spot which would be the same, but I understand the point that we should preserve `undefined` especially for those values that go into other selectors.